internal/configs/action.go

```diff
@@ -10,22 +10,37 @@ import (
 	"github.com/hashicorp/hcl/v2/hclsyntax"
 
 	"github.com/hashicorp/terraform/internal/addrs"
+	"github.com/hashicorp/terraform/internal/lang/langrefs"
+	"github.com/hashicorp/terraform/internal/tfdiags"
 )
 
+// There are many ways of handling plurality in error messages (linked_resource
+// vs linked_resources); this is one of them.
+type diagFn func(*hcl.Range) *hcl.Diagnostic
+
 func invalidLinkedResourceDiag(subj *hcl.Range) *hcl.Diagnostic {
 	return &hcl.Diagnostic{
 		Severity: hcl.DiagError,
-		Summary:  `Invalid "linked_resource"`,
```

Comment:
Out of curiosity and for my learning - why remove quotations?

There are more such instances in this very file and I a quick search shows that we use them for this purpose in other parts of the code.

Reply:
I was seeing those in tests where none of the other messages had quotes around the attribute, so I removed them, but you make a good point that actually did nothing for consistency.

Reply:
I'll add diagnostics to the (not-yet-extant) list of things that need style guides 😁